centreon-certified/bsm/bsm_connector-apiv2.lua

```diff
@@ -0,0 +1,300 @@
+    end
+
+    self.sc_logger:info(http_post_data)
+    return true
+  end
+  
+  local http_post_data = ""
+
+  for _, xml_str in ipairs(data) do
+    http_post_data = http_post_data .. tostring(xml.eval(xml_str))
+  end
+
+  self.sc_logger:info("[EventQueue:send_data]: Going to send the following json " .. tostring(http_post_data))
+  self.sc_logger:info("[EventQueue:send_data]: BSM Http Server URL is: \"" .. tostring(self.sc_params.params.http_server_url .. "\""))
+
+  local http_response_body = ""
+  local http_request = curl.easy()
+  :setopt_url(self.sc_params.params.http_server_url)
+  :setopt_writefunction(
+    function (response)
+      http_response_body = http_response_body .. tostring(response)
+    end
+  )
+  :setopt(curl.OPT_TIMEOUT, self.sc_params.params.connection_timeout)
+  :setopt(curl.OPT_SSL_VERIFYPEER, self.sc_params.params.allow_insecure_connection)
+  :setopt(
+    curl.OPT_HTTPHEADER,
+    {
+      "Content-Type: text/xml",
+      "content-length: " .. string.len(http_post_data)
+    }
+  )
+
+  -- set proxy address configuration
+  if (self.sc_params.params.proxy_address ~= '') then
+    if (self.sc_params.params.proxy_port ~= '') then
+      http_request:setopt(curl.OPT_PROXY, self.sc_params.params.proxy_address .. ':' .. self.sc_params.params.proxy_port)
+    else
+      self.sc_logger:error("[EventQueue:send_data]: proxy_port parameter is not set but proxy_address is used")
+    end
+  end
+
+  -- set proxy user configuration
+  if (self.sc_params.params.proxy_username ~= '') then
+    if (self.sc_params.params.proxy_password ~= '') then
+      http_request:setopt(curl.OPT_PROXYUSERPWD, self.sc_params.params.proxy_username .. ':' .. self.sc_params.params.proxy_password)
+    else
+      self.sc_logger:error("[EventQueue:send_data]: proxy_password parameter is not set but proxy_username is used")
+    end
+  end
+
+  -- adding the HTTP POST data
+  http_request:setopt_postfields(http_post_data)
+  -- performing the HTTP request
+  http_request:perform()
+  -- collecting results
+  http_response_code = http_request:getinfo(curl.INFO_RESPONSE_CODE)
+  http_request:close()
+
+  -- Handling the return code
+  local retval = false
+  if http_response_code == 202 or http_response_code == 200 then
+    self.sc_logger:info("[EventQueue:send_data]: HTTP POST request successful: return code is " .. tostring(http_response_code))
+    retval = true
+  else
+    self.sc_logger:error("[EventQueue:send_data]: HTTP POST request FAILED, return code is " .. tostring(http_response_code) .. ". Message is: " .. tostring(http_response_body))
+  end
+  return retval
+end
+
+--------------------------------------------------------------------------------
+-- Required functions for Broker StreamConnector
+--------------------------------------------------------------------------------
+
+local queue
+
+-- Fonction init()
+function init(conf)
+  queue = EventQueue.sc_event.event.new(conf)
```

Comment:
you are in the init() method, you want to init the EventQueue object, and not do some random thing (I know that is because you made a ctrl^H and it has replaced the original init :p)